### PR TITLE
Values and axis limits

### PR DESCRIPTION
**Pull request by Felixmil**
We can now decide to set limits on values or/and axis. see [these examples](https://ggplot2.tidyverse.org/reference/coord_cartesian.html#ref-examples) for explanations.

Setting limits on values can help reducing time for drawing plot while axis limits act just as a zoom on the plot.

### CODE REVIEW

File: R/plotconfiguration-axis.R [hunk 1]
```diff
@@ -314,6 +334,7 @@ XAxisConfiguration <- R6::R6Class(
       suppressMessages(
         plotObject <- plotObject +
           ggplot2::scale_x_continuous(
```
Comment (Felixmil) [inline R/plotconfiguration-axis.R:336]: `scale_` filters the data

**Review by pchelle — COMMENTED** [review]
@Felixmil 
Would it be okay if we keep and deprecate `limits` ? (linking the deprecated `limits` to `axisLimits`)
I am worried that some/many functions of the reporting engine are relying on this property and might break from this PR

**Comment by Felixmil** [comment]
> Would it be okay if we keep and deprecate `limits` ? (linking the deprecated `limits` to `axisLimits`)
> I am worried that some/many functions of the reporting engine are relying on this property and might break from this PR

Yes sure, should a message or warning appear to tell the user to use `axisLimits` instead of `limits` ?

**Comment by pchelle** [comment]
> > Would it be okay if we keep and deprecate `limits` ? (linking the deprecated `limits` to `axisLimits`)
> > I am worried that some/many functions of the reporting engine are relying on this property and might break from this PR
> 
> Yes sure, should a message or warning appear to tell the user to use `axisLimits` instead of `limits` ?

I think it may be more appropriate to throw a warning indeed. I'll create an RE issue to handle such warnings.

**Comment by Felixmil** [comment]
@pchelle Done in last commits

**Review by pchelle — APPROVED** [review]

**Pull request merged**